sn_networking/src/lib.rs

```diff
@@ -633,14 +633,25 @@ impl Network {
 
         let cmd_sender = self.swarm_cmd_sender.clone();
 
-        // Spawn a task to send the SwarmCmd and keep this fn sync
-        let _handle = tokio::spawn(async move {
-            if capacity == 0 {
+        if capacity == 0 {
+            if matches!(cmd, SwarmCmd::AddKeysToReplicationFetcher { .. }) {
+                // we can safely drop AddKeysToReplicationFetcher
+                // it should be reattempted in a few seconds and if we can cope we'll do it.
+                warn!(
+                    "SwarmCmd channel is full. Dropping AddKeysToReplicationFetcher: {:?}",
+                    cmd
+                );
+                return Ok(());
+            } else {
                 error!(
                     "SwarmCmd channel is full. Await capacity to send: {:?}",
                     cmd
                 );
             }
+        }
+
+        // Spawn a task to send the SwarmCmd and keep this fn sync
+        let _handle = tokio::spawn(async move {
```

Comment:
I think the whole `capacity check and send` shall be in a spawned thread.
i.e. the whole function will be like:
```rust
    fn send_swarm_cmd(&self, cmd: SwarmCmd) -> Result<()> {
        let cmd_sender = self.swarm_cmd_sender.clone();

        // Spawn a task to send the SwarmCmd and keep this fn sync
        let _handle = tokio::spawn(async move {
            let capacity = cmd_sender.capacity();
           if capacity == 0 {
               if matches!(cmd, SwarmCmd::AddKeysToReplicationFetcher { .. }) {
                    // we can safely drop AddKeysToReplicationFetcher
                    // it should be reattempted in a few seconds and if we can cope we'll do it.
                    warn!(
                        "SwarmCmd channel is full. Dropping AddKeysToReplicationFetcher: {:?}",
                        cmd
                    );
                    return Ok(());
                } else {
                    error!(
                        "SwarmCmd channel is full. Await capacity to send: {:?}",
                        cmd
                    );
                }
            }

            if let Err(error) = cmd_sender.send(cmd).await {
                error!("Failed to send SwarmCmd: {}", error);
            }
        });

        Ok(())
    }
```

Reply:
I wondered that, but i thought this would avoid what (im not sure how much) work there is in spawning threads there?